dsp/modules/google.py

```diff
@@ -0,0 +1,105 @@
+import math
+from typing import Any, Optional
+import backoff
+
+from dsp.modules.lm import LM
+
+try:
+    import google.generativeai as genai
+except ImportError:
+    google_api_error = Exception
+    print("Not loading Google because it is not installed.")
+
+def backoff_hdlr(details):
+    """Handler from https://pypi.org/project/backoff/"""
+    print(
+        "Backing off {wait:0.1f} seconds after {tries} tries "
+        "calling function {target} with kwargs "
+        "{kwargs}".format(**details)
+    )
+
+
+def giveup_hdlr(details):
+    """wrapper function that decides when to give up on retry"""
+    if "rate limits" in details.message:
+        return False
+    return True
+
+
+class Google(LM):
+    """Wrapper around Google's API.
+
+    Currently supported models include `gemini-pro-1.0`.
+    """
+
+    def __init__(
+        self,
+        model: str = "gemini-pro-1.0",
+        api_key: Optional[str] = None,
+        stop_sequences: list[str] = [],
+        **kwargs
+    ):
+        """
+        Parameters
+        ----------
+        model : str
+            Which pre-trained model from Google to use?
+            Choices are [`gemini-pro-1.0`]
+        api_key : str
+            The API key for Google.
+            It can be obtained from https://cloud.google.com/generative-ai-studio
+        **kwargs: dict
+            Additional arguments to pass to the API provider.
+        """
+        super().__init__(model)
+        self.google = genai.configure(api_key="")
+        self.provider = "google"
+        self.kwargs = {
+            "model_name": model,
+            "temperature": 0.0,
```

Comment:
`temperature` and other configuration might be convenient to update through `kwargs`

How about something like this?

```
"temperature": 0.0 if "temperature" not in kwargs else kwargs["temperature"],

```

Reply:
Awesome, good call!